### PR TITLE
Purge the last remnants of bsd.cheri.mk

### PR DESCRIPTION
**Pull request by jrtc27**
This requires removing one rtld test that currently only is built for hybrid MIPS, as our bsd.compat.mk doesn't let you build binaries without any CHERI support when it's present. It's a pretty uninteresting test though that only vaguely makes sense for MIPS where we have CHERI bits in the ELF flags (as opposed to just purecap bits).

### CODE REVIEW

**Comment by jrtc27** [comment]
Jenkins is happy, not that it really tests much given LIBCHERI is in __DEFAULT_NO_OPTIONS and cheribuild doesn't set it (with no nice way to do so, only `--cheribsd/build-options=-DWITH_LIBCHERI`).

**Review by brooksdavis — APPROVED** [review]

File: share/mk/bsd.cheri.mk [hunk 1]
```diff
@@ -1,78 +0,0 @@
...
-.endif
-
-.if ${WANT_CHERI} == "sandbox"
-# Force position-dependent sandboxes; PIEs aren't supported
-NO_SHARED=	yes
-.endif
-
-# Don't remove CHERI symbols from the symbol table
-STRIP_FLAGS+=	-w --keep-symbol=__cheri_callee_method.\* \
-		--keep-symbol=__cheri_method.\*
```
Comment (arichardson) [inline share/mk/bsd.cheri.mk:75]: Same here? Not that I see much value in keeping the libcheri prototype around...

File: bin/cheri_bench/Makefile [hunk 2]
```diff
@@ -2,11 +2,15 @@
 
 .include <src.opts.mk>
 
+.if !${MACHINE_ABI:Mpurecap}
+NEED_COMPAT=	CHERI
+.include <bsd.compat.mk>
```
Comment (arichardson) [inline bin/cheri_bench/Makefile:7]: Is this program ever used? If not I think we should just delete it.

**Review by arichardson — APPROVED** [review]
I think this is probably fine to merge as is just a few questions.

**Review by bsdjhb — APPROVED** [review]
I'm happy to see it go, though I think Alex asked some good questions.  I think the best way to address Alex's questions if they are issues is to remove more things. :)

**Comment by jrtc27** [comment]
Yeah; I suspect sandboxes have indeed been broken. I've been meaning to file the "remove libcheri" PR (though I suspect scattered cruft will remain for a while), and this should probably land after that.

**Comment by jrtc27** [comment]
(also then we can GC all the cheri_ccall/cheri_ccallee crud from LLVM :tada:)

File: share/mk/bsd.cheri.mk [hunk 3]
```diff
@@ -1,78 +0,0 @@
...
-# we will probably generate wrong code for calling them
-CFLAGS+=-Werror=implicit-function-declaration
-.endif
-.else
-STATIC_CFLAGS+= -ftls-model=local-exec # MIPS/hybrid case
-.endif
-
-.if ${WANT_CHERI} == "sandbox"
-# Force position-dependent sandboxes; PIEs aren't supported
-NO_SHARED=	yes
```
[the remark this answers is not in the document]
Comment (jrtc27) [reply share/mk/bsd.cheri.mk]: Gone with the libcheri removal

File: share/mk/bsd.cheri.mk (hunk 1 above)
Comment (jrtc27) [reply share/mk/bsd.cheri.mk]: Gone with the libcheri removal

File: bin/cheri_bench/Makefile (hunk 2 above)
Comment (jrtc27) [reply bin/cheri_bench/Makefile]: Gone with the libcheri removal

**Pull request merged**